apps/govquests-frontend/src/domains/notifications/components/NotificationToaster.tsx

```diff
@@ -0,0 +1,45 @@
+import { Toaster } from "@/components/ui/toaster";
+import { useToast } from "@/hooks/use-toast";
+import { useEffect } from "react";
+import { useNotificationDiff } from "../hooks/useNotificationDiff";
+import { useNotifications } from "../hooks/useNotifications";
+import { NotificationStrategyFactory } from "../strategies/NotificationStrategyFactory";
+import { NotificationNode } from "../types/notificationTypes";
+
+const NotificationToaster = () => {
+  const { data } = useNotifications();
+  const { toast } = useToast();
+
+  const notifications = data?.pages?.flatMap(
+    (value) => value.notifications.edges,
+  );
+  const { getNewNotifications } = useNotificationDiff(notifications);
+
+  useEffect(() => {
+    const newNotifications = getNewNotifications();
+
+    newNotifications.forEach((notification) => {
+      displayNotification(notification.node, toast);
+    });
+  }, [notifications]);
+
+  return <Toaster />;
+};
+
+export default NotificationToaster;
+
+const displayNotification = (
+  notification: NotificationNode,
+  toast: ReturnType<typeof useToast>["toast"],
+) => {
+  const notificationStrategy = NotificationStrategyFactory.createStrategy(
```

Comment:
I was reflecting a bit about this name, and I think it's a bit innacurate. We're actually reacting to a notification, and this naming might convey we're processing the notifications. I thought of perhaps using `OnNotificationHandler`, `OnNotificationHandlerFactory`, etc. WDYT?

Reply:
another two cents, but I think the dependencies for the classes that might be instantiated (namely, `toast`) should be passed to the factory not to the strategy itself because

Reply:
I think it will be easier to understand this way, I'll change!